src/Hocon.Configuration.Test/ConfigurationSpec.cs

```diff
@@ -594,11 +594,48 @@ public void WithFallback_ShouldNotChangeOriginalConfig()
             a.WithFallback(b);
 
             a.Fallbacks.Count.Should().Be(0);
-            a.GetString("akka.other-key").Should().BeNull();
+            a.GetString("akka.other-key", null).Should().BeNull();
             ReferenceEquals(oldA, a).Should().BeTrue();
             oldAContent.Should().Equals(a);
         }
 
+        [Fact]
+        public void WithFallback_ShouldMergeSubstitutionProperly()
```

Comment:
Ah, this is the reproduction case you were talking about. Good, glad we have it.